File: src/main/frontend/src/models/Joueur.ts
import {TournoiIndividuel} from "./TournoiIndividuel";

export class Joueur{
  id: number = null;
  organisateur: string = "";
  Tournois: Array<TournoiIndividuel> = new Array();
  nom: string = "";
  prenom: string = "";

  villeId: number = null;

  nomUtilisateur: string = "";
  email: string = "";
  adresse: string = "";
  pays: string = "";
  codePostale: number = null;
  apropos: string = "";
  citation: string = "";

  levele: number = 0;
  points: number = 0;

  newUser: boolean = true;
  active: boolean = false;
  confirm: boolean = false;

  photo: string = "";
  coverImg: string = "";

  created_at: Date = new Date;
  updated_at: Date = null;

}

File: src/main/java/esport/sprint1/sprint1/models/Joueur.java
package esport.sprint1.sprint1.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonSetter;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "Joueur")
public class Joueur extends Utilisateur implements Serializable {
    @ManyToMany(mappedBy = "joueurInscrit", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<TournoisIndividuel> tournois;
    public Joueur() {
    }

    public Joueur(String nom, String prenom, Long villeId, String nomUtilisateur, String email, String adresse, String pays, int codePostale, String apropos, String citation, int levele, int points, boolean newUser, boolean active, boolean confirm, String photo, String coverImg, Date created_at, Date updated_at, String password, List<TournoisIndividuel> tournois) {
        super(nom, prenom, villeId, nomUtilisateur, email, adresse, pays, codePostale, apropos, citation, levele, points, newUser, active, confirm, photo, coverImg, created_at, updated_at, password);
        this.tournois = tournois;
    }

    @JsonIgnore
    public List<TournoisIndividuel> getTournois() {
        return tournois;
    }

    @JsonSetter
    public void setTournois(List<TournoisIndividuel> tournois) {
        this.tournois = tournois;
    }
}
